app/models/article.rb

```diff
@@ -1,29 +1,38 @@
 # frozen_string_literal: true
 
 class Article < ApplicationRecord
+  TITLE_LENGTH_MAX = 50
   VALID_STATUSES = {
     draft: 0,
     published: 1,
     trashed: 2
   }.freeze
 
   LANGUAGES = {
-    ukr: 0,
-    eng: 1
+    ua: 0,
+    en: 1
   }.freeze
 
-  enum status:   VALID_STATUSES, _prefix: true
-  enum language: LANGUAGES, _prefix: true
+  enum status:   VALID_STATUSES
+  enum language: LANGUAGES
 
-  has_many    :article_tags
+  belongs_to  :user, foreign_key: 'user_id', inverse_of: :articles
+  has_many    :article_tags, dependent: :destroy
   has_many    :tags, through: :article_tags
-  belongs_to  :user
 
-  default_scope -> { order(created_at: :desc) }
+  validates   :title, :text, :user, :status, :language, presence: true
+  validates   :title, length: { maximum: TITLE_LENGTH_MAX }
 
-  validates   :title, presence: true, length: { maximum: 50 }
-  validates   :text, presence: true
-  validates   :user, presence: true
-  validates   :status, presence: true
-  validates   :language, presence: true
+  # This is required for the following function in another PR
+  # scope :sorted_desc, -> { order(created_at: :desc) }
+
+  def tag_list=(tags_string)
+    tag_names = tags_string.split(',').collect { |s| s.strip.downcase }.uniq
+    new_or_found_tags = tag_names.collect { |name| Tag.find_or_create_by(name: name) }
+    self.tags = new_or_found_tags
+  end
+
+  def tag_list_names
```

Comment:
Use presenter for this, it's forbidden to put such code in model.

Reply:
@mmrshk  this method moved in presenter